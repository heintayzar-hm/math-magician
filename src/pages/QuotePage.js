import React from 'react';

// eslint-disable-next-line react/prefer-stateless-function
class QuotePage extends React.Component {
  render() {
    const style = {
      h2: {
        fontSize: '25px',
        fontWeight: '700',
      },
      main: {
        height: '80vh',
        width: '80vw',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        margin: 'auto',
      },
    };
    return (
      <>
        <main style={style.main}>
          <h2 style={style.h2}>
            Mathematics is not about calculation, equations, computations, or algorithms,
            It is about understanding --- Willian Paul Thurston
          </h2>
        </main>
      </>
    );
  }
}

export default QuotePage;
