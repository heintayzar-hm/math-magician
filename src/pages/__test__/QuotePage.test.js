import React from 'react';
import { render } from '@testing-library/react';
import QuotePage from '../QuotePage';

describe('QuotePage', () => {
  it('renders correctly', () => {
    // Arrange

    // Act
    const { container } = render(<QuotePage />);

    // Assert
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the correct quote text', () => {
    // Arrange

    // Act
    const { container } = render(<QuotePage />);
    const quote = container.querySelector('h2');

    // Assert
    expect(quote.textContent).toBe('Mathematics is not about calculation, equations, computations, or algorithms, It is about understanding --- Willian Paul Thurston');
  });
});
