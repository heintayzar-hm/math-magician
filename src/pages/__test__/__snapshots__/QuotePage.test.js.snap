// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotePage renders correctly 1`] = `
<main
  style="height: 80vh; width: 80vw; display: flex; align-items: center; justify-content: center; margin: auto;"
>
  <h2
    style="font-size: 25px; font-weight: 700;"
  >
    Mathematics is not about calculation, equations, computations, or algorithms, It is about understanding --- Willian Paul Thurston
  </h2>
</main>
`;
